src/transformer_deploy/backends/onnx_utils.py

```diff
@@ -0,0 +1,236 @@
+#  Copyright 2022, Lefebvre Dalloz Services
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import copy
+import sys
+from collections import defaultdict
+from pathlib import Path
+
+import numpy as np
+import onnx
+from onnx import ModelProto
+from onnx.shape_inference import infer_shapes_path
+
+
+def save_onnx(proto: onnx.ModelProto, model_path: str) -> None:
+    """
+    Save Onnx file with external data support when required.
+
+    :param proto: Onnx model
+    :param model_path: output path
+    """
+    # protobuff doesn't support files > 2Gb, in this case, weights are stored in another binary file
+    save_external_data: bool = proto.ByteSize() > 2 * 1024**3
+    filename = Path(model_path).name
+    onnx.save_model(
```

Comment:
We can't add some assertion to be sure model is well saved ?

Reply:
Why do you think it may fail? 
We may add a unit tests if you think it's relevant.

Reply:
We can add unittest if it is possible (test it is working fine and it is the right path).